contract-examples/test/tx_allow_list.ts

```diff
@@ -1,199 +1,128 @@
+
+    console.log('')
+  }
+
+  assert(!failed, `${fnName} failed`)
 }
 
+const test = (name, fnName, overrides = {}) => it(name, testFn(fnName, overrides));
+test.only = (name, fnName, overrides = {}) => it.only(name, testFn(fnName, overrides));
+test.debug = (name, fnName, overrides = {}) => it.only(name, testFn(fnName, overrides, true));
+
 describe("ExampleTxAllowList", function () {
-  let admin: SignerWithAddress
-  let contract: Contract
-  let allowed: SignerWithAddress
-  let noRole: SignerWithAddress
-  before(async function () {
-    admin = await ethers.getSigner(adminAddress)
-    const contractF: ContractFactory = await ethers.getContractFactory("ExampleTxAllowList", { signer: admin })
-    contract = await contractF.deploy()
-    await contract.deployed()
-    const contractAddress: string = contract.address
-    console.log(`Contract deployed to: ${contractAddress}`)
-
-      ;[, allowed, noRole] = await ethers.getSigners()
-
-    // Fund allowed address
-    await admin.sendTransaction({
-      to: allowed.address,
-      value: ethers.utils.parseEther("10")
-    })
-
-    // Fund no role address
-    let tx = await admin.sendTransaction({
-      to: noRole.address,
-      value: ethers.utils.parseEther("10")
-    })
-    await tx.wait()
-  })
+  beforeEach('Setup DS-Test contract', async function () {
+    const signer = await ethers.getSigner(ADMIN_ADDRESS)
+    const allowListPromise = ethers.getContractAt("IAllowList", TX_ALLOW_LIST_ADDRESS, signer);
 
-  it("should add contract deployer as admin", async function () {
-    const contractOwnerAdmin: string = await contract.isAdmin(contract.owner())
-    expect(contractOwnerAdmin).to.be.true
+    return ethers.getContractFactory("ExampleTxAllowListTest", { signer })
+      .then(factory => factory.deploy())
+      .then(contract => {
+        this.testContract = contract;
+        return contract.deployed().then(() => contract)
+      })
+      .then(contract => contract.setUp())
+      .then(tx => Promise.all([allowListPromise, tx.wait()]))
+      .then(([allowList]) => allowList.setAdmin(this.testContract.address))
+      .then(tx => tx.wait())
   })
 
-  it("precompile should see admin address has admin role", async function () {
-    // test precompile first
-    const allowList = await ethers.getContractAt("IAllowList", TX_ALLOW_LIST_ADDRESS, admin)
-    let adminRole = await allowList.readAllowList(admin.address)
-    expect(adminRole).to.be.equal(ROLES.ADMIN)
-  })
+  test("should add contract deployer as admin", "contractOwnerIsAdmin")
 
-  it("precompile should see test address has no role", async function () {
-    // test precompile first
-    const allowList = await ethers.getContractAt("IAllowList", TX_ALLOW_LIST_ADDRESS, admin)
-    let role = await allowList.readAllowList(noRole.address)
-    expect(role).to.be.equal(ROLES.NONE)
-  })
+  // this is testing something that is setup in the before hook 
+  // I don't really understand why... 
```

Comment:
I don't think adding assertions to the `setUp` functions would make for very clear errors. If for whatever reason the deployer was not setup as an admin properly, most of the tests would fail and it should be clear-ish that something is seriously wrong.

I don't have a super strong opinion here though, if you think I should add error-logging (for assertions) to the `setUp` block, I can do that.